r/tests/testthat/test-expression.R

```diff
@@ -18,25 +18,44 @@
 context("Expressions")
 
 test_that("Can create an expression", {
-  expect_is(Array$create(1:5) + 4, "Expression")
+  expect_is(Array$create(1:5) + 4, "array_expression")
 })
 
 test_that("Recursive expression generation", {
   a <- Array$create(1:5)
-  expect_is(a == 4 | a == 3, "Expression")
+  expect_is(a == 4 | a == 3, "array_expression")
 })
 
-test_that("as.vector(expression)", {
+test_that("as.vector(rexpression)", {
   a <- Array$create(1:5)
   expect_equal(as.vector(a + 4), 5:9)
   expect_equal(as.vector(a == 4 | a == 3), c(FALSE, FALSE, TRUE, TRUE, FALSE))
 })
 
-test_that("Expression print method", {
+test_that("array_expression print method", {
   a <- Array$create(1:5)
   expect_output(
     print(a == 4 | a == 3),
     capture.output(print(c(FALSE, FALSE, TRUE, TRUE, FALSE))),
     fixed = TRUE
   )
 })
+
+test_that("C++ expressions", {
+  f <- FieldExpression$create("f")
```

Comment:
`field_expr("f")` seems more readable.

Reply:
`<class>$create()` is what we settled on, but sometimes we have snake case underscore factory functions. No strong view on this, but at least `<class>$create()` is easily recognized as a constructor call.

Reply:
I don't expect humans to make field expressions directly, so that's why there's no convenience constructor.

Reply:
ah ok, if it's not exposed to user, than ignore my comment.